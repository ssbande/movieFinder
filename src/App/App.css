body {
  background-color: #333;
  color: white;
  font-family: Lato,'Helvetica Neue',Arial,Helvetica,sans-serif !important;
}

.app {
  padding: 20px;
}

.appHeading {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 40px;
  cursor: pointer;
}

.flexInline {
  display: flex;
  justify-content: space-between;
  align-items: center;
}