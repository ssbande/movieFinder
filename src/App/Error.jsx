import React from 'react';
import { Message } from 'semantic-ui-react';

const Error = () => {
  return (
    <Message negative>
      <Message.Header>We're sorry something has gone wrong</Message.Header>
      <p>Please try searching again ...</p>
    </Message>
  )
}

export default Error;