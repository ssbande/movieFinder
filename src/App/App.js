import React from 'react';
import { withRouter, Switch, Route } from 'react-router-dom';
import { Header, Divider } from 'semantic-ui-react';

import MovieDetails from '../Details/Movies';
import ShowDetails from '../Details/Shows';
import PersonDetails from '../Details/Persons';
import Home from '../Landing/Home';
import SearchBar from '../Search/SearchBar';
import constants from '../utils/constants';
import { HistoryProps } from '../utils/propTypes';

import 'semantic-ui-css/semantic.min.css'
import './App.css';

const App = props => {
  const { history } = props;
  const { appHeading, config: { movie, tv, person } } = constants;

  return (
    <div className='app'>
      <div className='appHeading'>
        <Header id="heading" as='h1' color='teal' onClick={() => history.push('/')}>{appHeading}</Header>
      </div>
      <Divider inverted />
      <SearchBar />
      <Switch>
        <Route exact path='/'><Home /></Route>
        <Route path='/movies/:movieId'><MovieDetails category={movie.category} /></Route>
        <Route path='/shows/:showId'><ShowDetails category={tv.category} /></Route>
        <Route path='/artists/:artistId'><PersonDetails category={person.category} /></Route>
      </Switch>
    </div>
  );
}

App.propTypes = {
  history: HistoryProps
}

export default withRouter(App);
