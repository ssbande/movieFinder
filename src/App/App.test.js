import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

const setup = () => {
  return {
    props: {
      history: {
        push: jest.fn
      }
    }
  }
}

test('render App properly', () => {
  const { props } = setup();
  const wrapper = shallow(<App {...props} />);
  expect(wrapper).toMatchSnapshot();
})