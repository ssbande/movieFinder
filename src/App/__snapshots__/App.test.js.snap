// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App properly 1`] = `ShallowWrapper {}`;
